src/main/java/core/basesyntax/service/parsing/FileParserImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.parsing;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+public class FileParserImpl implements FileParser {
+    private static final String DATA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE = 0;
+    private static final int FRUIT_TYPE = 1;
+    private static final int QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> parse(List<String> records) {
+        return records.stream()
+                .map(r -> r.split(DATA_SEPARATOR))
+                .map(record -> new FruitTransaction(operationIdentity(record[OPERATION_TYPE]),
+                        record[FRUIT_TYPE], Integer.parseInt(record[QUANTITY])))
+                .collect(Collectors.toList());
+    }
+
+    private FruitTransaction.Operation operationIdentity(String operation) {

```

Comment:
```suggestion
    private FruitTransaction.Operation getOperation(String operation) {
```
common comment: please use verbs for method names